src/unix.rs

```diff
@@ -21,13 +21,27 @@ pub struct Acquired {
 }
 
 impl Client {
-    pub fn new(limit: usize) -> io::Result<Client> {
+    pub fn new(mut limit: usize) -> io::Result<Client> {
+        assert!(limit <= 4096, "Can't have limit larger than 4096 since the pipe would block forever when writing to it");
+
         let client = unsafe { Client::mk()? };
+
         // I don't think the character written here matters, but I could be
         // wrong!
-        for _ in 0..limit {
-            (&client.write).write_all(&[b'|'])?;
+        //
+        // Also, it is highly unlikely to have a machine with more than
+        // 128 cores.
+        const BUFFER: [u8; 128] = [b'|'; 128];
+
+        while limit >= BUFFER.len() {
```

Comment:
If you'd like this could be tightened up a bit with:

```
while limit > 0 {
    (&client.write).write_all(&BUFFER[..limit.min(BUFFER.len())])?;
    limit -= BUFFER.len();
}
```